libraries/AP_RangeFinder/AP_RangeFinder_Backend_Serial.h

```diff
@@ -31,4 +31,9 @@ class AP_RangeFinder_Backend_Serial : public AP_RangeFinder_Backend
 
     // maximum time between readings before we change state to NoData:
     virtual uint16_t read_timeout_ms() const { return 200; }
+
+    // 0 is no return value, 100 is perfect.  false means signal_quality
+    // is not available and quality_pct set to SIGNAL_QUALITY_NODATA.
+    virtual bool get_signal_quality_pct(int8_t &quality_pct) const WARN_IF_UNUSED
```

Comment:
change to use saem signature as the frontend


i.e. return int8_t and UNKNOWN if no datam not bool ean

Reply:
Good idea. Changed the return value. Left the name slightly different (get_signal_quality_pct() vs signal_quality_pct()) to prevent a name collision of these two different functions. get_signal_quality_pct() is called by the update routine to populate the state structure. While signal_quality_pct() is called by a client to get values from the state structure.